evalml/pipelines/pipeline_base.py

```diff
@@ -709,7 +709,7 @@ def repr_component(parameters):
         parameters_repr = ", ".join(
             [
                 f"'{component}':{{{repr_component(parameters)}}}"
-                if component != DFSTransformer.name
+                if DFSTransformer.name not in component
```

Comment:
This is another way of checking for dfs transformers in a pipeline than the `isinstance` checks above, right? When should one be used over the other?

At some point, I feel like it might be worth rethinking how component names are used across automl search. We have multiple ways that a component's name in a component graph doesn't match the actual name of the component itself (numeric vs categorical pipelines and stacked ensemble pipelines are the two that come to mind but there may be more), and it's easy to have bugs like these. 

It'd be nice if there was a more clear way to distinguish between "I want to check something for any DFS Transformer component in a pipeline" vs "I want to check an individual DFS Transformer instance in a pipeline". Not sure what that'd look like, but maybe it's worth making an issue that we can take on at some point in the future.

Reply:
Maybe something as simple as `element.component_graph.get_components(DFSTransformer)` that returns a list of all components of that type

Reply:
I definitely agree with you @tamargrey and thought about the same when I was writing up this PR. In this particular case, the component names are strings so I had to do this string check but I followed your example for checking for DFS below using `has_dfs = any(isinstance(c, DFSTransformer) for c in element.component_graph)`. I'll take a look into how we can standardize it.

Reply:
@tamargrey took a look and we use `has_dfs = any(isinstance(c, DFSTransformer) for c in self.component_graph)` across the board except here where I need to use the component name. Does that work for you?

Reply:
yep, that makes sense to me! Maybe just add a comment here explaining why we have to use the component name instead of the other way